### ISSUE TITLE
[BUG] playwright object and browse object cannot be reused

### ISSUE THREAD

**Issue by ylhang** [microsoft/playwright-java#689]
**Context:**
- Playwright Version: 1.15.2
- Operating System: windows 10
- Browser: Chromium

In order to speed up the processing speed of Playwright, I created a blockqueue to consume incoming crawl requests

```java
while (true) {
                try {
                    req = blockingQueue.poll(2, TimeUnit.SECONDS);
                    if (null == req) {
                        Thread.sleep(200);
                        continue;
                    }
                    Field messageField = playwright.getClass().getDeclaredField("driverProcess");
                    messageField.setAccessible(true);
                    Process proc = (Process)messageField.get(playwright);
                    if (!proc.isAlive()){
                        log.info("playwright is dead");
                        closeBrowse(browser);
                        closePlaywright(playwright);
                        playwright = create();
                    }
                    String reqUrlStr = req.getReqUrls().stream().collect(Collectors.joining(" ; "));
                    log.info(Thread.currentThread().getName() + "\t consume" + reqUrlStr + "success" + "\t\t" + "resource queue= " + blockingQueue.size());
                    List<String> reqUrls = req.getReqUrls();
                    String browse = req.getBrowse();
                    if (StringUtils.isEmpty(browse)) {
                        browse = "chromium";
                    }

                    BrowserType browserType = getBrowserType(playwright, browse);
                    if (browser == null || !browser.isConnected()) {
                        closeBrowse(browser);
                        //browserContext = browserType.launchPersistentContext(path, new BrowserType.LaunchPersistentContextOptions().setTimeout(60000).setHeadless(false));
                        browser = browserType.launch(new BrowserType.LaunchOptions().setHeadless(true));
                    }

                    List<List<String>> partition = ListUtils.partition(reqUrls, 10);
                    for (List<String> reqUrlsPartition:partition) {
                        browserContext = browser.newContext(new Browser.NewContextOptions().setIgnoreHTTPSErrors(true));
                        String result = crawlPage(tag, reqUrlsPartition, browserContext);
                        log.info(tag + ",V3 Consume Finish......." + result);
                    }
                } catch (Throwable ex) {
                    log.error(tag + ",consume error", ex);
                } finally {
                    Thread.sleep(5);
                }
            }
```

**Describe the bug**
1. When I request multiple times but do not close the browse object, only close browserContext in method crawlPage, 
 I may report the error that the context cannot be created, the error code occur in the statement
```java
browserContext = browser.newContext(new Browser.NewContextOptions().setIgnoreHTTPSErrors(true));
```
2. When I forcibly kill the Node process in the background, even though I judged and recreated PlayWright in the code, it actually reported PlayWright is closed.

The question is : 
How can I reuse PlayWright and Browse objects, because the resources and time required to restart them every time will be very huge

**Comment by ylhang** [comment]
we are using spring boot ,the main code contains:
DemoApplication.java：The spring boot entry startup program, which contains the command to start the consumer
TestPlaywrightController.java：http api, you can initiate a fetching call through a GET request
ShareDataV3.java :  Specific logic implementation class
pom.xml： maven dependencies

You can access: http://localhost:8080/api/crawl/get?reqUrls=&browse=chromium&headless=true to trigger the fetching operation
==Another serious problem: when i use webkit driver , the code will be dead , i don't know the reason==


The relevant documents are as follows：===github seems to be unable to send attachments, so I will post the code directly===
```java
package com.study.demo;

import com.study.demo.utils.ShareDataV4;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@SpringBootApplication
public class DemoApplication implements CommandLineRunner {

    public static void main(String[] args) {
        SpringApplication.run(DemoApplication.class, args);
    }

    @Override
    public void run(String... args) throws Exception {
        ExecutorService executorService = Executors.newFixedThreadPool(5);
        for (int i = 0; i <2 ; i++) {
            final int k=i;
            CompletableFuture.runAsync(new Runnable() {
                @Override
                public void run() {
                    try {
                        ShareDataV4.consume("consumer"+k);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
            },executorService);
        }

    }
}
```

**Comment by yury-s** [comment]
What is the error you are getting?

It may well be that the browser dies while you are crawling some of the pages, can you collect some logs by starting the process with `DEBUG=pw:browser` and see if the browser crashes? 

Try to reduce your example, otherwise it's hard to follow.

**Comment by ylhang** [comment]
1. when i force close node process after palywright created, it will trigger error always  ,     but  browse connected property is true after i had close node process
![image](https://user-images.githubusercontent.com/13460226/140445492-02d580fa-35bb-4893-ac50-a2b4cf08384a.png)


2.  when i change driver to webkit ,it will be dead in the follow "page.navigate(reqUrl);" ,there are no log.
```java
            browserContext.setDefaultTimeout(10000);
            browserContext.setDefaultNavigationTimeout(2000);

            for (int i = 0; i < reqUrls.size(); i++) {
                Page page = null;
                try {
                    page = browserContext.newPage();
                    //Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/97.0.4666.0 Safari/537.36
                    //Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36
                    page.addInitScript("navigator.webdriver = false; Object.defineProperty(navigator, 'webdriver', {get: () => false});Object.defineProperty(navigator, 'userAgent', {get: () => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36'});");
                    String reqUrl = reqUrls.get(i);
                    page.navigate(reqUrl);

                } catch (Exception ex) {
                } finally {
                    pages.add(page);
                }
            }
//...................................
```

**Comment by yury-s** [comment]
> 1. when i force close node process after palywright created, it will trigger error always  ,     but  browse connected property is true after i had close node process

What do you mean by "force close node process", do you kill the process instead of using playwright.close() ?


> 2. when i change driver to webkit ,it will be dead in the follow "page.navigate(reqUrl);" ,there are no log.
> 
There are should be at least some logs, when you run with `DEBUG=pw:browser` and the browser launches, can you share those?

**Comment by yury-s** [comment]
We need more information to act on this report. Please file a new one and link to this issue when you get back to it!
